ee/vellum_ee/workflows/server/virtual_file_loader.py

```diff
@@ -1,42 +1,72 @@
 import importlib
+import re
+from typing import Optional
 
 
 class VirtualFileLoader(importlib.abc.Loader):
-    def __init__(self, code: str, is_package: bool):
-        self.code = code
-        self.is_package = is_package
+    def __init__(self, files: dict[str, str], namespace: str):
+        self.files = files
+        self.namespace = namespace
 
     def create_module(self, spec):
         return None  # use default module creation
 
     def exec_module(self, module):
-        if not self.is_package or self.code:
-            exec(self.code, module.__dict__)
+        module_info = self._resolve_module(module.__spec__.origin)
 
+        if module_info:
+            file_path, code = module_info
+            compiled = compile(code, file_path, "exec")
+            exec(compiled, module.__dict__)
 
-class VirtualFileFinder(importlib.abc.MetaPathFinder, importlib.abc.Loader):
-    def __init__(self, files: dict[str, str], namespace: str):
-        self.files = files
-        self.namespace = namespace
+    def get_source(self, fullname):
+        """
+        `inspect` module uses this method to get the source code of a module.
+        """
+
+        module_info = self._resolve_module(fullname)
+        if module_info:
+            return module_info[1]
+
+        return None
+
+    def _resolve_module(self, fullname: str) -> Optional[tuple[str, str]]:
```

Comment:
Most of the logic from `VirtualFileFinder.find_spec` was moved to here, so that if could work with any `inspect` call invoked within the module